### DISCUSSION TITLE
Changing Object Identifier using /samples/ReadWriteProperty.py

### DISCUSSION THREAD

**Discussion by owenwu-ACT** [JoelBender/bacpypes#495] [Q&A]
Hi folks,
I am new to bacpypes and trying to explore the functionalities of the stack. I was trying to change a device's object ID (network 500 address 3 object id 9999) to 999. I first tried using ReadWriteProperty.py: write 500:3 device:9999 objectIdentifier 999 but got an error "ValueError: invalid format". I then tried WriteSomething.py using: write 500:3 8:9999 75 999 but got an error "NameError: name 'deferred' is not defined'.

Has anyone done it before and let me know what I have done wrong? Thank you so very mcuh!

**Comment by JoelBender** [comment]
The `ValueError` is correct, you are trying to write a integer/unsigned to an object identifier property.  Try this:
```
> write 500:3 device:9999 objectIdentifier device:999
```
The `deferred()` error is a bug.
